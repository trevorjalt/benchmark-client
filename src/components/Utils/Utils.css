.Input {
  border: 1px solid #ddd;
  display: block;
  font-size: 1.1em;
  margin-bottom: 10px; 
  padding: 1vh 0 1vh 0;
  text-align: left;
  width: 95vw;
  border-radius: 4px;
}

.Input::placeholder {
  color: #ccc;
}

.Button {
  order: 1px solid #aaa;
  background: white;
  font-size: 1.1em;
  border-radius: 20px;
  border-color: purple;
  padding: 8px 12px;
  font-weight: bold;
}

.Required {
  color: #D11053;
  font-family: Arial;
  font-weight: bold;
}

.Section {
  margin-left: 2.5vw;
  margin-right: 2.5vw;
  width: 90vw;
}

