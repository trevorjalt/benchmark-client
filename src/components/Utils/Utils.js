import React from 'react'
import { format as formatDate } from 'date-fns'
import './Utils.css'

export function NiceDate({ date, format='do MMMM yyyy' }) {
    return formatDate(date, format)
}

export function Button({ className, ...props }) {
    return <button className={['Button', className].join(' ')} {...props} />
}

export function Input({ className, ...props }) {
    return (
        <input className={['Input', className].join(' ')} {...props} />
    )
}

export function Required({ className, ...props }) {
    return (
        <span className={['Required', className].join(' ')} {...props}>
            &#42;
        </span>
    )
}

export function Section({ className, ...props }) {
    const classes = [
        'Section',
        className,
    ].filter(Boolean).join(' ')
    return (
        <section className={classes} {...props} />
    )
}