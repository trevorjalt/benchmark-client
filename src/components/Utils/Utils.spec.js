import React from 'react'
import { NiceDate, Button, Input, Required, Section } from './Utils'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import './Utils.css'

describe(`Utils Component`, () => {
    const dateProps = {
        date: new Date(),
        format: 'do MMMM yyyy',
    }

    const props = {
        className: 'LimitBreaker',
    }
    
    it (`returns a nice date`, () => {
        const wrapper = 
            shallow (
                <NiceDate {...dateProps} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it (`returns a button`, () => {
        const wrapper = 
            shallow (
                <Button {...props} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it (`returns an input field`, () => {
        const wrapper = 
            shallow (
                <Input {...props} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it (`returns required`, () => {
        const wrapper = 
            shallow (
                <Required {...props} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it (`returns a section`, () => {
        const wrapper = 
            shallow (
                <Section {...props} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })  
})