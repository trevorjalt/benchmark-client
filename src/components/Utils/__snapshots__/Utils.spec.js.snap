// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils Component returns a button 1`] = `
<button
  className="Button LimitBreaker"
/>
`;

exports[`Utils Component returns a nice date 1`] = `"15th October 2020"`;

exports[`Utils Component returns a section 1`] = `
<section
  className="Section LimitBreaker"
/>
`;

exports[`Utils Component returns an input field 1`] = `
<input
  className="Input LimitBreaker"
/>
`;

exports[`Utils Component returns required 1`] = `
<span
  className="Required LimitBreaker"
>
  *
</span>
`;
